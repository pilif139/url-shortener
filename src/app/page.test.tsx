import {render, screen} from "@testing-library/react";
import "@testing-library/jest-dom"
import Home from "@/app/page";
import {jest} from "@jest/globals";

describe("Home Component", () => {
  it("should render correctly", () => {
    const consoleErrorSpy = jest.spyOn(console, "error").mockImplementation(()=>{}); //jest have problem with next.js server actions so it throws error when we try to render the component with server actions
    const {container} = render(<Home/>);
    expect(container).toMatchSnapshot();
    consoleErrorSpy.mockRestore();
  });
});