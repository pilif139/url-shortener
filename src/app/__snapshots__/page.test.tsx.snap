// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render correctly 1`] = `
<div>
  <main
    class="flex flex-col items-center justify-center text-2xl"
  >
    <form
      class="flex flex-col items-center gap-5"
    >
      <input
        class="rounded-2xl p-2 outline-none transition w-[15em] focus:bg-slate-100 dark:placeholder:text-slate-800 dark:bg-slate-400 dark:text-black md:w-[30em] dark:focus:bg-slate-300"
        name="url"
        placeholder="Enter your url here..."
        style="opacity: 0; will-change: opacity,transform; transform: translateX(-200px);"
        type="text"
      />
      <input
        class="rounded-2xl p-2 outline-none transition w-[15em] focus:bg-slate-100 dark:placeholder:text-slate-800 dark:bg-slate-400 dark:text-black md:w-[30em] dark:focus:bg-slate-300"
        name="alias"
        placeholder="Your optional alias..."
        style="opacity: 0; will-change: opacity,transform; transform: translateX(200px);"
        type="text"
      />
      <button
        class="mx-4 w-max rounded-2xl bg-violet-500 p-2 px-8 text-white transition hover:bg-violet-600"
        style="opacity: 0; will-change: opacity,transform; transform: scale(0.9);"
        type="submit"
      >
        Submit
      </button>
    </form>
  </main>
</div>
`;
